### PR TITLE
Fix crash on startup without connection to opc ua server

### PR DESCRIPTION
**Pull request by OParczyk**
Fixes crash on startup, if opc server could not be connected to.

### CODE REVIEW

File: src/processHistorian.py [hunk 1]
```diff
@@ -112,11 +110,6 @@ def __init__(self):
         print("Work threads:")
         print(self.__threads)
 
-        # Eighth step: Subscribe all data and start all threads
-        self._opcua_client.subscribe_all()
-        for thread in self.__threads:
-            thread.start()
```
Comment (C0ntroller) [inline src/processHistorian.py:118]: Removing this also makes the buffer push not working

File: src/processHistorian.py [hunk 2]
```diff
@@ -112,11 +110,6 @@ def __init__(self):
         print("Work threads:")
         print(self.__threads)
 
-        # Eighth step: Subscribe all data and start all threads
-        self._opcua_client.subscribe_all()
```
Comment (C0ntroller) [inline src/processHistorian.py:116]: Im not sure if it's a good idea to not setup correnctly but wait for an error

**Review by C0ntroller — CHANGES_REQUESTED** [review]

File: src/processHistorian.py (hunk 2 above)
Comment (OParczyk) [reply src/processHistorian.py:116]: Suppose the watchdog works - we'd then have a structure which deals well with "no connection"-States - Startup really is just a special form of no-connection
The alternative would be to replicate most of the functionality of the watchdog just for the initial connection.

File: src/processHistorian.py (hunk 1 above)
Comment (OParczyk) [reply src/processHistorian.py:118]: That is true and a big bug, thanks!

File: src/processHistorian.py [hunk 3]
```diff
@@ -85,7 +83,6 @@ def __init__(self):
 
         # Fifth step: Create the OPC UA client
         self._opcua_client = OPCClient(self.__opcua_conf, self._buffer.append)
-        self._opcua_client.connect()
```
Comment (C0ntroller) [inline src/processHistorian.py:88]: This does not work. 
connect() also initializes the lists for polling and subscribing. Without this the polling threads will not be created.

Maybe opcuaClient.Client.__init_lists should be edited so no connection is necessary?

**Review by C0ntroller — CHANGES_REQUESTED** [review]

File: src/processHistorian.py (hunk 3 above)
Comment (OParczyk) [reply src/processHistorian.py:88]: connect is called by the defibrillator. As namespace ids may have changed during restart of opc reinitializing lists on (re)connect is a good thing IMO

Comment (OParczyk) [reply src/processHistorian.py:88]: Oh I see

Comment (C0ntroller) [reply src/processHistorian.py:88]: Thats not the point. 
It is absolutely necessary to initialize the dictionaries in this code segment as the workthreads are based on the list of keys in the polling dictionary. Else the list of keys would be empty and no Threads will be created

Comment (OParczyk) [reply src/processHistorian.py:88]: We cant reliably do that without an ensured connection though. Maybe store the push thread separately (there is only one) and recreate the opc threads from scratch on reconnect.
Sidenote: what happenes, if a node to be polled doesn't exist?

Comment (OParczyk) [reply src/processHistorian.py:88]: This should (tm) now work reliably, given no thread runs away with the lock

**Review by C0ntroller — APPROVED** [review]

**Pull request merged**